### PR TITLE
[fix][broker]Fix mutex never released when trimming

### PR DESCRIPTION
**Pull request by 315157973**
### Motivation
During the Ledger trimming process, if the number of BK nodes is not enough or the BK cluster is temporarily unavailable, the new Ledger will fail to be created.
At this point, the `TreeMap` will return null, and `firstNonDeletedLedger` is the primitive type, so an NPE will be thrown, and the mutex will never be released

image

### Modifications
Throw and catch the exception

### Verifying this change
Unit test can pass
https://github.com/315157973/pulsar/pull/1

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [x] `doc-not-needed` 
bug fix

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by lhotari — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by HQebupt — APPROVED** [review]

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java [hunk 1]
```diff
@@ -2652,15 +2659,19 @@ public void operationFailed(MetaStoreException e) {
...
             return;
         }
 
         // need to move mark delete for non-durable cursors to the first ledger NOT marked for deletion
         // calling getNumberOfEntries latter for a ledger that is already deleted will be problematic and return
         // incorrect results
-        long firstNonDeletedLedger = ledgers.higherKey(ledgersToDelete.get(ledgersToDelete.size() - 1).getLedgerId());
+        Long firstNonDeletedLedger = ledgers.higherKey(ledgersToDelete.get(ledgersToDelete.size() - 1).getLedgerId());
+        if (firstNonDeletedLedger == null) {
+            throw new ManagedLedgerNotFoundException("First non deleted Ledger is not found");
```
Comment (codelipenghui) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java:2673]: A little confused for `ManagedLedgerNotFoundException`
I think `LedgerNotExistException` is better for this case and easier to understand.

**Review by codelipenghui — COMMENTED** [review]
Great Job!

**Review by coderzc — APPROVED** [review]

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java (hunk 1 above)
Comment (315157973) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java]: Done

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**